activerecord/lib/active_record/relation/delegation.rb

```diff
@@ -100,7 +100,17 @@ def #{method}(...)
              :to_sentence, :to_fs, :to_formatted_s, :as_json,
              :shuffle, :split, :slice, :index, :rindex, to: :records
 
-    delegate :primary_key, :connection, :transaction, to: :klass
+    delegate :primary_key, :connection, :table_name, :transaction, :arel_table, :uncached,
+             :sanitize_sql_like, :unscoped, to: :klass
+
+    # TODO: scoped delegate
+    [:find_signed, :find_signed!, :delete, :find_by_token_for, :find_by_token_for!, :upsert_all, :insert_all, :insert_all!].each do |method|
```

Comment:
This list is very likely incomplete.

Reply:
So I rebased, and this is still the main blocker. We have a number of `ActiveRecord::Base` class methods that rely on this scoped delegation.

I need to find a good alternative.